maths/is_palindrome.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * A function to see if a number is a Palindrome.
+ *
+ * @param number The input number.
+ * @return {boolean} Wether the number is a Palindrome or not.
+ * @example isPalindrome(12321) => true | isPalindrome(455) => false
+ * @see https://en.wikipedia.org/wiki/Palindromic_number
+ * @author FatimaChariwala <https://github.com/fatima-0000>
+ */
+
+export const IsPalindrome = (number: number): boolean => {
+    const stringValue = number.toString();
+    const length = stringValue.length - 1;
+
+    for (let i = 0; i <= length; i++) {
```

Comment:
This algorithm can be made more efficient by only checking till the $\lfloor length/2 \rfloor$-th index, which can be explained by a comment.

Reply:
if we really want efficiency, we should spare ourselves the `toString()` and just iterate over the digits

Reply:
Currently, the code converts the number to a string. This is wasteful. You can directly iterate over the digits of `x` using `x % 10` and `floor(x / 10)`. Similarly, you can iterate over the digits in reverse order. This is what I have in mind:

```typescript
// Assumption: x is an integer >= 1.
const isPalindromic = (x: number) => {
    let msdpow = 10**Math.log10(x)
    let lsdpow = 1
    while (msdpow > lsdpow) {
        const msd = Math.floor(x / msdpow) % 10 -- most significant digit
        const lsd = Math.floor(x / lsdpow) % 10 -- least significant digit
        if (msd != lsd) return false;
        msdpow /= 10 -- "remove" MSD
        lsdpow *= 10 -- "remove" LSD
    }
    return true;
}
```

Reply:
Your code doesn't seem to work.
What about this one?
`if (number < 0 || (number % 10 === 0 && number !== 0)) {
        return false;
    }

    let reversed = 0;
    while (number > reversed) {
        reversed = reversed * 10 + (number % 10);
        number = Math.floor(number / 10);
    }

    return number === reversed || number === Math.floor(reversed / 10);`

Reply:
The while-loop which reverses the number works as following:
1. Extract the last digit of the number (`x % 10`) and add this digit to the right end of `reversed` variable.
2. Remove the last digit from the number (`Math.floor(x / 10)`)

Reply:
> Your code doesn't seem to work.

Fixed (I didn't write it with editor support or test it, it was just for illustrative purposes, should've made that more clear).

> What about this one?

Yeah, that's the approach the JavaScript implementation uses. It has logarithmic time complexity as well. It is probably also more efficient than what I proposed.